rows.go

```diff
@@ -126,6 +126,69 @@ func checkTypeValue(columnType string, val interface{}) error {
 	return fmt.Errorf("unknown column type: %s", columnType)
 }
 
+func extractStructColumn(columnType string) (string, string, error) {
+	field := strings.SplitN(strings.TrimSpace(columnType), " ", 2)
+	if len(field) != 2 {
+		return "", "", fmt.Errorf("invalid struct field: %s", columnType)
+	}
+	return strings.TrimSpace(field[0]), strings.TrimSpace(field[1]), nil
+}
+
+func extractStructColumns(columnTypes string) (map[string]string, error) {
+	balance := 0
+	current := strings.Builder{}
+	columns := make(map[string]string)
+	for _, char := range columnTypes {
+		if char == '(' {
+			balance++
+		} else if char == ')' {
+			balance--
+		}
+		if balance == 0 && char == ',' {
+			fieldName, fieldType, err := extractStructColumn(current.String())
+			if err != nil {
+				return nil, err
+			}
+			columns[fieldName] = fieldType
+			current.Reset()
+		} else {
+			current.WriteRune(char)
+		}
+	}
+	if balance != 0 {
+		return nil, fmt.Errorf("invalid struct type: %s", columnTypes)
+	}
+	fieldName, fieldType, err := extractStructColumn(current.String())
+	if err != nil {
+		return nil, err
+	}
+	columns[fieldName] = fieldType
+	return columns, nil
+}
+
+func parseStruct(structInnerFields string, val interface{}) (map[string]driver.Value, error) {
+	fields, err := extractStructColumns(structInnerFields)
+	if err != nil {
+		return nil, ConstructNestedError("error during parsing struct type", err)
+	}
+	structValue, ok := val.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("unexpected value for struct type: %v", val)
+	}
+	res := make(map[string]driver.Value)
+	if len(fields) != len(structValue) {
+		return nil, fmt.Errorf("expected %d fields, but got %d", len(fields), len(structValue))
+	}
+	for fieldName, fieldType := range fields {
+		if fieldValue, ok := structValue[fieldName]; ok {
+			res[fieldName], _ = parseValue(fieldType, fieldValue)
```

Comment:
Let's add a comment that we're ignoring error on purpose here

Reply:
Good that you've notices since we actually shouldn't ignore it here :)